docs/guides/databases_with_concord.md

```diff
@@ -2,14 +2,134 @@
+In this guide, you will either decide on choosing SQLite3 to start, or postgreSQL, which is the most recommended one for production.
 
-## Opening the DB file
+## PostgreSQL
 
-The first step in this guide will be opening the DB file, as it's required for the other sqlite3 functions like reading and writing data.
+PostgreSQL is a really advanced database, and incredibly fast, but it's hard to use it, at least, compared to easier databases, like SQLite3.
+
+### Opening the DB file
+
+Before executing any type of function on the database, you will first need to connect to it, which luckily is incredibly easy to do.
+
+See an example below, where we are opening the DB file, and checking if it failed to open.
+
+```c
+#include <libpq-fe.h> // PostgreSQL library
+
+...
+
+/* Here we connect to the database, and store the connection in the conn struct. */
+PGconn *conn = PQconnectdb("host=POSTGRESQL-HOSTNAME port=5432 dbname=POSTGRESQL-DBNAME user=POSTGRESQL-USER-NAME password=POSTGRESQL-PASSWORD");
+
+/* Checking if the connection failed, if it did, it will log the error message, and deallocate the conn struct with PQfinish. */
+switch (PQstatus(conn)) {
+    case CONNECTION_OK:
+    case CONNECTION_MADE:
+        /* Successfully connected, you can proceed using database functions. */
+        log_trace("[libpq] Successfully connected to the postgres database server.");
+        break;
+    case CONNECTION_STARTED:
+        /* Still connecting, but really soon you will be connected. */
+        log_trace("[libpq] Waiting for connection to be made.");
+        break;
+    case CONNECTION_AWAITING_RESPONSE:
+        /* Waiting for a response from the database, probably will respond soon, if properly configured. */
+        log_trace("[libpq] Waiting for a response from the server.");
+        break;
+    case CONNECTION_AUTH_OK:
+        /* Authentication was successful, but the backend is still starting up. */
+        log_trace("[libpq] Received authentication; waiting for backend start-up to finish.");
+        break;
+    case CONNECTION_SSL_STARTUP:
+        /* Negotiating SSL encryption, since the connected database was using SSL encryption. */
+        log_trace("[libpq] Almost connecting; negotiating SSL encryption.");
+        break;
+    case CONNECTION_SETENV:
+        /* Negotiating environment-driven parameter settings, soonly will be connected */
+        log_trace("[libpq] Almost connecting; negotiating environment-driven parameter settings.");
+        break;
+    default:
+        /* Ugh, something went wrong while connecting to the database, probably a wrong password, firewall, or something blocking the connection. */
+        log_fatal("[libpq] Error when trying to connect to the postgres database server. [%s]\n", PQerrorMessage(conn));
+        PQfinish(conn);
+}
+```
+
+### Creating a table
+
+We got the database connected, so now we can now proceed creating a table, since it's required before saving any data into it.
```

Comment:
We got connected to the database, so now we can now proceed creating a table. We need one since it's the only way you can actually store data in a SQL DB.